x-pack/plugins/remote_clusters/public/sections/remote_cluster_list/components/disconnect_button/disconnect_button.js

```diff
@@ -40,48 +41,103 @@ export class DisconnectButtonUi extends Component {
     });
   };
 
+  onConfirm = () => {
+    const { disconnectClusters, clusterNames } = this.props;
+    disconnectClusters(clusterNames);
+    this.closeConfirmModal();
+  }
+
+  renderButtonText() {
+    const { clusterNames, isSmallButton } = this.props;
+    const isSingleCluster = clusterNames.length === 1;
+
+    if(isSmallButton) {
+      return (
+        <FormattedMessage
+          id="xpack.remoteClusters.disconnectButton.shortButtonLabel"
+          defaultMessage="Disconnect"
+        />
+      );
+    }
+
+    if(isSingleCluster) {
+      return (
+        <FormattedMessage
+          id="xpack.remoteClusters.disconnectButton.singleButtonLabel"
+          defaultMessage="Disconnect remote cluster"
+        />
+      );
+    }
+
+    return (
+      <FormattedMessage
+        id="xpack.remoteClusters.disconnectButton.multipleButtonLabel"
+        defaultMessage="Disconnect {count} remote clusters"
+        values={{ count: clusterNames.length }}
+      />
+    );
+  }
+
   render() {
-    const { disconnectRemoteClusters, clusterNames, intl } = this.props;
+    const { intl, clusterNames } = this.props;
     const { isModalOpen } = this.state;
+    const isSingleCluster = clusterNames.length === 1;
 
     let modal;
 
     if (isModalOpen) {
+      const title = isSingleCluster ? intl.formatMessage({
+        id: 'xpack.remoteClusters.disconnectButton.confirmModal.deleteSingleClusterTitle',
+        defaultMessage: 'Disconnect remote cluster \'{name}\'?',
+      }, { name: clusterNames[0] }) : intl.formatMessage({
+        id: 'xpack.remoteClusters.disconnectButton.confirmModal.multipleDeletionTitle',
+        defaultMessage: 'Disconnect {count} remote clusters?',
+      }, { count: clusterNames.length });
+
```

Comment:
Personally I read better the code above like this

```js
const title = isSingleCluster
  ? intl.formatMessage({
    id: 'xpack.remoteClusters.disconnectButton.confirmModal.deleteSingleClusterTitle',
    defaultMessage: 'Disconnect remote cluster \'{name}\'?',
  }, { name: clusterNames[0] })
  : intl.formatMessage({
    id: 'xpack.remoteClusters.disconnectButton.confirmModal.multipleDeletionTitle',
    defaultMessage: 'Disconnect {count} remote clusters?',
  }, { count: clusterNames.length });
```
but that's personal, up to you 😊